modules/cloudevent-trigger/main.tf

```diff
@@ -78,7 +78,18 @@ resource "google_pubsub_subscription" "this" {
     }
   }
 
-  expiration_policy {
-    ttl = "" // This does not expire.
+  dynamic "expiration_policy" {
+    for_each = [var.expiration_policy]
+    content {
+      ttl = expiration_policy.value.ttl
+    }
+  }
+
+  dynamic "retry_policy" {
+    for_each = [var.retry_policy]
+    content {
```

Comment:
FWIW `for_each` over a single element isn't really useful, this can be simplified.